File: frontend/src/types/match.ts
export interface MatchResult {
  profile_id: string | null;
  company_name: string;
  industry_sector?: string;
  operating_regions?: string[];
  match_score: number | null;
  explanation?: string;
  locked: boolean;
}

export interface MatchListResponse {
  matches: MatchResult[];
  total: number;
  has_more: boolean;
}

File: backend/app/schemas/match.py
import uuid
from typing import Optional, List
from pydantic import BaseModel


class MatchResult(BaseModel):
    profile_id: Optional[uuid.UUID]
    company_name: str
    industry_sector: Optional[str] = None
    operating_regions: Optional[List[str]] = None
    match_score: Optional[float]
    explanation: Optional[str] = None
    locked: bool = False


class MatchListResponse(BaseModel):
    matches: List[MatchResult]
    total: int
    has_more: bool
